src/imageview.cpp

```diff
@@ -161,15 +165,22 @@ void ImageView::mouseReleaseEvent(QMouseEvent* event) {
                         textRect.top() + (textRect.height() / 2 - radius),
                         radius * 2, radius * 2);
 
-      // Draw the path
+      // Draw the path in the image
       QPainterPath path;
       path.addEllipse(circleRect);
       painter.fillPath(path, Qt::red);
       painter.drawPath(path);
+      // Draw the path in the sence
+      annotations.append(scene_->addPath(path, painter.pen(), QBrush(Qt::red)));
 
-      // Draw the text
+      // Draw the text in the image
       painter.setPen(Qt::white);
       painter.drawText(textRect, Qt::AlignCenter, text);
+      // Draw the text in the sence
+      QGraphicsTextItem *textItem = scene_->addText(text, painter.font());
+      textItem->setPos(textRect.x() - textRect.width() / 8 , textRect.y() - textRect.height() / 8);
```

Comment:
Please avoid magical numbers! 8 is an arbitrary integer here.

The reason why the text isn't centered is that `QGraphicsTextItem` supports HTML text and Qt has an old issue when dealing with the position of HTML texts, incorrectly considering their HTML tags when positioning them.

Use `QGraphicsSimpleTextItem` instead. Then, these 5 lines (from the comment to `annotations.append(textItem);`) can be rewritten as:
```c++
      // Draw the text in the sence
      QGraphicsSimpleTextItem *textItem = new QGraphicsSimpleTextItem(text);
      textItem->setFont(font);
      textItem->setBrush(Qt::white);
      textItem->setPos(textRect.topLeft());
      scene_->addItem(textItem);
      annotations.append(textItem);
```